src/cdk/stepper/stepper.ts

```diff
@@ -120,6 +120,16 @@ export class CdkStep implements OnChanges {
     this._stepper.selected = this;
   }
 
+  /** Resets the step to its initial state. Note that this includes resetting form data. */
+  reset(): void {
+    this.interacted = false;
+    this.completed = false;
+
+    if (this.stepControl) {
+      this.stepControl.reset();
```

Comment:
Note: I'm open to having this reset only the `dirty` and `touched` states. I went with `reset`, because leaving behind the value didn't seem like a "true" reset.

Reply:
This seems like the most common use case to me. If we get  lot of requests for reset all but the value we could create a new method: `{reset,clear}{Status,State}`